context/app/api/client.py

```diff
@@ -177,11 +177,17 @@ def get_vitessce_conf_cells_and_lifted_uuid(self, entity, marker=None, wrap_erro
             # TODO: Entity structure will change in the future to be consistent
             # about "files". Bill confirms that when the new structure comes in
             # there will be a period of backward compatibility to allow us to migrate.
-            derived_entity['files'] = derived_entity['metadata'].get('files', [])
-            vitessce_conf = self.get_vitessce_conf_cells_and_lifted_uuid(
-                derived_entity, marker=marker, wrap_error=wrap_error
-            ).vitessce_conf
-            vis_lifted_uuid = derived_entity['uuid']
+
+            metadata = derived_entity.get('metadata', {})
+
+            if (metadata.get('files')):
+                derived_entity['files'] = metadata.get('files', [])
+                vitessce_conf = self.get_vitessce_conf_cells_and_lifted_uuid(
+                    derived_entity, marker=marker, wrap_error=wrap_error).vitessce_conf
+                vis_lifted_uuid = derived_entity['uuid']
+            else:  # no files
+                vitessce_conf = _create_vitessce_error(
+                    'No files found in lifted image pyramid descendants metadata.')
```

Comment:
Do end users know what "lifted image pyramid descendants" are? 

If we're showing an error message to the user, should this also be logged? Perhaps at Info level, not Error.

Reply:
> 'Related entity {uuid} is missing files information in its metadata'

This is a lot better message for end users. (delete the 's' after 'files' for grammar, but optional since I know you have the 's' for the `files` key), but my only question is for people debugging the issue, is there a way for them to know from that message (or any other similar fast ways) that the `lifted_image_pyramid`is missing metadata?

Reply:
It makes sense to conform to proper grammar so things don't look wrong to end users - I get put off when I see incorrect grammar on a site. We should also be able to further increase specificity.

Fresh adjustment: `'Related image entity {uuid} is missing information (no "files" key found in its metadata).'`